cli/src/command.rs

```diff
@@ -221,6 +202,25 @@ fn ensure_dev(spec: &Box<dyn service::ChainSpec>) -> std::result::Result<(), Str
 	}
 }
 
+/// Runs a performance check via compiling sample wasm code with a timeout.
+/// Should only be run in release build since the check would take too much time otherwise.
+/// Returns `Ok` immediately if the check has been passed previously.
+fn host_perf_check() -> Result<()> {
+	#[cfg(build_type = "debug")]
+	{
+		Err(PerfCheckError::DebugBuildNotSupported.into())
+	}
+	#[cfg(build_type = "release")]
+	{
+		crate::host_perf_check::host_perf_check().map_err(Into::into)
+	}
+	#[cfg(not(any(build_type = "debug", build_type = "release")))]
+	{
+		info!("Performance check skipped: unknown build type");
+		Ok(())
+	}
```

Comment:
I would argue we don't want to support this, and hence this is also an error case.